src/lxc/start.c

```diff
@@ -1386,6 +1386,12 @@ static int lxc_spawn(struct lxc_handler *handler)
 		goto out_delete_net;
 	}
 
+	if (!lxc_list_empty(&conf->procs)) {
+		ret = setup_proc_filesystem(&conf->procs, handler->pid);
```

Comment:
I like that you placed it in `lxc_spawn()`. This runs after `lxc_setup()` has already finished in the child, if I remember correctly. If so then this is fine.